tool/release.rb

```diff
@@ -103,7 +103,12 @@ def extra_entry
     private
 
     def bundler_version
-      version.segments.map.with_index {|s, i| i == 0 ? s - 1 : s }.join(".")
+      v = if version.segments[0] >= 4
+        version
+      else
+        version.segments.map.with_index {|s, i| i == 0 ? s - 1 : s }.join(".")
+      end
+      v.to_s.gsub(/([a-z])\.(\d)/i, '\1\2')
```

Comment:
This may be better if we don't want to change the `< 4` case:

```suggestion
      if version.segments[0] >= 4
        version.to_s.gsub(/([a-z])\.(\d)/i, '\1\2')
      else
        version.segments.map.with_index {|s, i| i == 0 ? s - 1 : s }.join(".")
      end
```